include/deal.II/base/mpi.h

```diff
@@ -1124,6 +1124,20 @@ namespace Utilities
               const T &          object_to_send,
               const unsigned int root_process = 0);
 
+    /**
+     * A function that combines values @p local_value from all processes
+     * via a user-specified binary operation @p combiner on the @p root_process.
+     * As such this function is similar to MPI_Reduce but due to the
+     * user-specified binary operation also general object types, including
+     * ones that store variable amounts of data, can be handled.
+     */
```

Comment:
I think it would be good if you explained the difference between this and the `all_reduce` function below.